.github/workflows/publish.yml

```
@@ -20,30 +51,28 @@ jobs:
     needs: [build]
     steps:
       - name: Checkout code
-        uses: actions/checkout@v4
+        uses: actions/checkout@v2
```

Is the version downgrade intentional? (Same for line 56,58 and 75)
```suggestion
        uses: actions/checkout@v4
```